modules/extensions-mule-support/src/main/java/org/mule/runtime/module/extension/mule/internal/loader/parser/MuleSdkParameterModelParserSdk.java

```diff
@@ -44,23 +52,83 @@ public class MuleSdkParameterModelParserSdk extends BaseMuleSdkExtensionModelPar
 
   protected final ComponentAst parameterAst;
   private final TypeLoader typeLoader;
+  private final ExtensionModelHelper extensionModelHelper;
 
   private String name;
+  private MetadataType type;
+  private List<StereotypeModel> allowedStereotypes = emptyList();
+  private Optional<ParameterDslConfiguration> parameterDslConfiguration = empty();
```

Comment:
Note that now we are going to be precalculating all these when instantiating the parser. This is similar to what we have been doing with other parsers so far. Also, consider that `ModelParsers` don't have a `parse` method.
This will make it easier to avoid repeated computations for the same value without needing to do a lazy value approach. The downside (which can actually be an upside depending how you look at it) is that parsing errors will be raised earlier (for example: even if the `type` is never queried). But I think that is actually a good thing in this context.